### PR TITLE
add formal API for APIserver cover/chaff requests from devices

### PR DESCRIPTION
**Pull request by mikehelmick**

### CODE REVIEW

File: pkg/controller/cover/cover.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+func (c *CoverAPI) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	var request api.CoverRequest
+	if err := controller.BindJSON(w, r, &request); err != nil {
+		c.logger.Errorf("failed to bind request: %v", err)
+		controller.WriteJSON(w, http.StatusBadRequest, api.Error("invalid request: %v", err))
+		return
+	}
+
+	// Don't immediately return.
```
Comment (squee1945) [inline pkg/controller/cover/cover.go:85]: Can you also say why?

File: pkg/controller/cover/cover.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	if err != nil {
+		c.logger.Errorf("error getting random int, using min sleep: %v", err)
+		delay = big.NewInt(0)
+	}
+
+	length := time.Duration(minDelayMS+delay.Int64()) * time.Millisecond
+	time.Sleep(length)
+}
+
+func (c *CoverAPI) randomData() (string, error) {
```
Comment (crwilcox) [inline pkg/controller/cover/cover.go:64]: Is there any way we can ensure these get configured well? If they aren't accurate to the live payload, the chaff is easily detected as chaff.

File: pkg/controller/cover/cover.go (hunk 1 above)
Comment (mikehelmick) [reply pkg/controller/cover/cover.go:85]: updated comment

**Comment by crwilcox** [comment]
/lgtm /hold

Comments are all about clarification so shouldn't require further review.

File: pkg/controller/cover/cover.go (hunk 2 above)
Comment (mikehelmick) [reply pkg/controller/cover/cover.go:64]: Yes - will send a followup that tracks and normalizes both request size and latency

https://github.com/mikehelmick/go-chaff

Comment (mikehelmick) [reply pkg/controller/cover/cover.go:64]: added necessary TODOs

**Pull request merged**